geomesa-cassandra/geomesa-cassandra-datastore/src/main/scala/org/locationtech/geomesa/cassandra/index/CassandraAttributeIndex.scala

```diff
@@ -9,13 +9,106 @@
 
 package org.locationtech.geomesa.cassandra.index
 
-import org.locationtech.geomesa.cassandra.data._
-import org.locationtech.geomesa.cassandra.index.CassandraIndexAdapter.ScanConfig
-import org.locationtech.geomesa.cassandra.{RowRange, RowValue}
-import org.locationtech.geomesa.index.index.AttributeIndex
-
-case object CassandraAttributeIndex
-    extends AttributeIndex[CassandraDataStore, CassandraFeature, Seq[RowValue], Seq[RowRange], ScanConfig]
-    with CassandraAttributeLayout with CassandraFeatureIndex with CassandraIndexAdapter  {
+import java.nio.ByteBuffer
+
+import org.locationtech.geomesa.cassandra.data.{CassandraDataStore, CassandraFeature}
+import org.locationtech.geomesa.cassandra.{NamedColumn, RowRange, RowValue}
+import org.locationtech.geomesa.index.index.IndexKeySpace._
+import org.locationtech.geomesa.index.index.ShardStrategy.NoShardStrategy
+import org.locationtech.geomesa.index.index.attribute.{AttributeIndex, AttributeIndexKey, AttributeIndexKeySpace, AttributeIndexValues}
+import org.locationtech.geomesa.index.index.{IndexKeySpace, ShardStrategy}
+import org.locationtech.geomesa.index.strategies.AttributeFilterStrategy
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+case object CassandraAttributeIndex extends CassandraAttributeIndex
+
+trait CassandraAttributeIndex extends CassandraTieredFeatureIndex[AttributeIndexValues[Any], AttributeIndexKey]
+    with AttributeFilterStrategy[CassandraDataStore, CassandraFeature, Seq[RowValue]] {
+
+  private val Index     = NamedColumn("attrIdx",   0, "smallint", classOf[Short],  partition = true)
+  private val Value     = NamedColumn("attrVal",   1, "text",     classOf[String])
+  private val Secondary = NamedColumn("secondary", 2, "blob",     classOf[ByteBuffer])
+  private val FeatureId = CassandraFeatureIndex.featureIdColumn(3)
+  private val Feature   = CassandraFeatureIndex.featureColumn(4)
+
+  override val name: String = AttributeIndex.Name
+
   override val version: Int = 2
```

Comment:
Just to confirm, V2 is correct, right?

Reply:
yeah, the change in the attribute index is that the shard byte and the attribute identifier bytes were flipped position. since c* already had those as separate columns, i didn't need to change the data format.